dolphinscheduler-alert/dolphinscheduler-alert-server/src/main/java/org/apache/dolphinscheduler/alert/service/AlertEventPendingQueue.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.alert.service;
+
+import org.apache.dolphinscheduler.alert.config.AlertConfig;
+import org.apache.dolphinscheduler.alert.metrics.AlertServerMetrics;
+import org.apache.dolphinscheduler.dao.entity.Alert;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class AlertEventPendingQueue extends AbstractEventPendingQueue<Alert> {
+
+    public AlertEventPendingQueue(AlertConfig alertConfig) {
+        super(alertConfig.getSenderParallelism() * 3 + 1);
```

Comment:
May I ask why we set `n * 3 + 1` here?

Reply:
Since we want to make the producer block if the consumer consumes is very slow, so we set a max value here.
And if the max value is slow e.g. consumer number, then the producer will be block/notify frequently. Increasing the queue capacity can reduce the frequency, It's just a number I like better.